tooling/minimize/intrinsics/src/lib.rs

```diff
@@ -29,3 +34,112 @@ pub unsafe fn deallocate(ptr: *mut u8, size: usize, align: usize) {
     let layout = Layout::from_size_align(size, align).unwrap();
     unsafe { System.deallocate(ptr, layout); }
 }
+
+static THREAD_HANDLES: Mutex<Vec<Option<JoinHandle<()>>>> = Mutex::new(Vec::new());
+
+// Spawn returns the index+1 of the thread because we don't have a JoinHandle for thread 0.
+pub fn spawn<F>(f: F) -> usize
+where F: FnOnce() -> () + Send + 'static,
+{
+    let handle = thread::spawn(f);
+    let mut vec = THREAD_HANDLES.lock().unwrap();
+    vec.push(Some(handle));
+    vec.len()
+}
+
+pub fn join(index: usize) {
+    if index == 0 { panic!("Can not join thread 0.") }
+
+    let mut vec = THREAD_HANDLES.lock().unwrap();
+    let join_handle = vec.get_mut(index-1);
+
+    if let Some(join_handle) = join_handle {
+        let join_handle = mem::replace(join_handle, None);
```

Comment:
This can be written a lot shorter. I'd start with

If you use `vec.get_mut(...).unwrap_or_else(|| panic!("joining non-existing thread")).take()`, then you have one level of nesting less and don't need the `replace` later.

Then you can also release the lock immediately again.